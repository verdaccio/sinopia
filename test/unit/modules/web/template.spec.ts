import renderTemplate from '../../../../src/api/web/html/template';

const manifest = require('./partials/manifest/manifest.json');

const exampleManifest = {
  css: ['main.css'],
  js: ['runtime.js', 'main.js'],
  ico: '/static/foo.ico',
};

describe('template', () => {
  test('custom render', () => {
    expect(
      renderTemplate(
        { options: { base: 'http://domain.com' }, manifest: exampleManifest },
        manifest
      )
    ).toMatchSnapshot();
  });

  test('custom title', () => {
    expect(
      renderTemplate(
        { options: { base: 'http://domain.com', title: 'foo title' }, manifest: exampleManifest },
        manifest
      )
    ).toMatchSnapshot();
  });

  test('custom title', () => {
    expect(
      renderTemplate(
        { options: { base: 'http://domain.com', title: 'foo title' }, manifest: exampleManifest },
        manifest
      )
    ).toMatchSnapshot();
  });

  test('custom logo', () => {
    expect(
      renderTemplate(
        {
          options: { base: 'http://domain.com', logo: 'http://domain/logo.png' },
          manifest: exampleManifest,
        },
        manifest
      )
    ).toMatchSnapshot();
  });

  test('meta scripts', () => {
    expect(
      renderTemplate(
        {
          options: { base: 'http://domain.com' },
          metaScripts: [`<style>.someclass{font-size:10px;}</style>`],
          manifest: exampleManifest,
        },
        manifest
      )
    ).toMatchSnapshot();
  });

  test('custom body after', () => {
    expect(
      renderTemplate(
        {
          options: { base: 'http://domain.com' },
          scriptsBodyAfter: [`<script src="foo"/>`],
          manifest: exampleManifest,
        },
        manifest
      )
    ).toMatchSnapshot();
  });

  test('custom body before', () => {
    expect(
      renderTemplate(
        {
          options: { base: 'http://domain.com' },
          scriptsbodyBefore: [`<script src="fooBefore"/>`, `<script src="barBefore"/>`],
          manifest: exampleManifest,
        },
        manifest
      )
    ).toMatchSnapshot();
  });
});
