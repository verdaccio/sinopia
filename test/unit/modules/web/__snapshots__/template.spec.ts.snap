// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template custom body after 1`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title></title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com"}
        </script>
        
      </head>    
      <body class="body">
      
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        <script src="foo"/>
      </body>
    </html>
  "
`;

exports[`template custom body before 1`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title></title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com"}
        </script>
        
      </head>    
      <body class="body">
      <script src="fooBefore"/><script src="barBefore"/>
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        
      </body>
    </html>
  "
`;

exports[`template custom logo 1`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title></title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com","logo":"http://domain/logo.png"}
        </script>
        
      </head>    
      <body class="body">
      
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        
      </body>
    </html>
  "
`;

exports[`template custom render 1`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title></title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com"}
        </script>
        
      </head>    
      <body class="body">
      
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        
      </body>
    </html>
  "
`;

exports[`template custom title 1`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title>foo title</title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com","title":"foo title"}
        </script>
        
      </head>    
      <body class="body">
      
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        
      </body>
    </html>
  "
`;

exports[`template custom title 2`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title>foo title</title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com","title":"foo title"}
        </script>
        
      </head>    
      <body class="body">
      
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        
      </body>
    </html>
  "
`;

exports[`template meta scripts 1`] = `
"
    <!DOCTYPE html>
      <html lang="en-us"> 
      <head>
        <meta charset="utf-8">
        <base href="http://domain.com">
        <title></title>        
        <link rel="icon" href="http://domain.com-/static/favicon.ico"/>
        <meta name="viewport" content="width=device-width, initial-scale=1" /> 
        <script>
            window.__VERDACCIO_BASENAME_UI_OPTIONS={"base":"http://domain.com"}
        </script>
        <style>.someclass{font-size:10px;}</style>
      </head>    
      <body class="body">
      
        <div id="root"></div>
        <script defer="defer" src="http://domain.com-/static/runtime.9be80fd172e81558124c.js"></script><script defer="defer" src="http://domain.com-/static/main.9be80fd172e81558124c.js"></script>
        
      </body>
    </html>
  "
`;
